### DISCUSSION TITLE
BlueRetro 1Player Adapter for PS1/2

### DISCUSSION THREAD

**Discussion by Micha213** [darthcloud/BlueRetro#26] [Show and tell]
I built a BlueRetro 1 Player Adapter for the PS1. For this I used an ESP32 d1 mini board and created a 3d printed housing. I tried to make everything as small as possible. The USB socket for updating the firmware is below the controller connection. In addition, I wired a 5V voltage converter. The housing is held together with four m2 screws and the controller connection was clamped between the upper and lower part of the housing.
The result so far looks like this:
![20210404_143100](https://user-images.githubusercontent.com/46410328/115130756-747a3480-9ff2-11eb-9a25-a10449db8156.jpg)
![20210404_143448](https://user-images.githubusercontent.com/46410328/115130761-7ba14280-9ff2-11eb-9f54-6542b34a336f.jpg)
![20210404_143422](https://user-images.githubusercontent.com/46410328/115130763-7fcd6000-9ff2-11eb-8f0c-09db51593c5d.jpg)
![20210404_143659](https://user-images.githubusercontent.com/46410328/115130765-82c85080-9ff2-11eb-8f6f-6609f729a1a6.jpg)

Now I have designed a shield for the ESP32 d1 mini. It makes wiring easier, offers space for an LDO, makes both LEDs and buttons accessible from the outside and fits into the 3d printed housing.
![brpcb](https://user-images.githubusercontent.com/46410328/115130740-47c61d00-9ff2-11eb-9ca6-76668ee62067.png)

I still have to order the PCB and see if everything works and have to make a few improvements to the housing. If everything works, I will make the data for the PCB and housing available for download.

**Comment by darthcloud** [comment]
Very nice, well done!

Some suggestions:
Since this is targeted for PSX/2 you could remove the universal FW pads, there really no point in using that FW on a specific board.

I would change the pinout labels to match BlueRetro documentation.

I know those labels are used everywhere else but they are just wrong. Official Sony schematic for both PSX and PS2 use the name I used in the docs.

**Comment by Micha213** [comment]
Ok, thanks for your suggestions.

I changed the pin labeling. I found it a bit confusing at the beginning with the labeling, because you can always find a different labeling on google.

I also removed the solder jumpers for the Universal FW. Is it ok if I still put the pins on GND or is it better to leave them free? Both FW versions should still run on GND.
![brpcb2](https://user-images.githubusercontent.com/46410328/115151197-1c801400-a06c-11eb-90d5-7ca4acdb2088.png)

**Comment by darthcloud** [comment]
Yeah GND them, you can also put player 2 IO5 (P2_DTR), IO26 (P2_SCK) & IO27 (P2_TXD) to 3.3V.

**Comment by Micha213** [comment]
I had already connected the three pins to + 3.3V beforehand. ;)

**Comment by Micha213** [comment]
The boards for the ESP32 d1 mini have now arrived. To be precise, this is the second revision of it. The first one worked, but the footpring for the d1 mini was about 1mm too far forward.
The new circuit board fits perfectly in the 3d molded housing.

I uploaded the data for the 3d printed housing to thingiverse:
https://www.thingiverse.com/thing:4835926

The board can be ordered from OSH Park or the Gerber data can be downloaded there:
https://oshpark.com/shared_projects/aW1v2Ei3

I have to add a little more information to the text on thingiverse.
![20210528_172718](https://user-images.githubusercontent.com/46410328/120099167-6ca5c800-c13a-11eb-8072-68884f8be6d9.jpg)
![20210528_173114](https://user-images.githubusercontent.com/46410328/120099170-70394f00-c13a-11eb-98e1-29702216d104.jpg)
![20210528_173152](https://user-images.githubusercontent.com/46410328/120099173-73343f80-c13a-11eb-9b06-c405dbf23cae.jpg)
![20210528_174845](https://user-images.githubusercontent.com/46410328/120099176-75969980-c13a-11eb-8b39-abedb31784c0.jpg)

**Reply by jim150** [reply]
I’m trying to build this adapter and I don’t quite get where to solder the pin5 of the psx connector (red cable).
I also have the problem that I’m unable to program one of my two ESP32 (it keeps waiting for connection) while the other works without problems.
![EB0DF627-3566-4968-8C30-AC16D454C83B](https://user-images.githubusercontent.com/4740182/124673307-ead35800-deb8-11eb-8948-4c20102f2da5.jpeg)

**Reply by darthcloud** [reply]
Dont connect it leave it like that.

Sometimes the auto program circuit don't work well, while the app is trying to programming it, hold EN button (and keep holding), then hold boot button (and keep holding, release EN button, then release boot button. That should put ESP32 in program mode. You may need to try a few times.

**Reply by jim150** [reply]
Thank you! I will leave it unconnected.

I will try it some more times to program with this method. Thanks!

**Reply by Micha213** [reply]
The red cable is for 3.5V of the PS1. This is not required for the shield.

I also had problems with some of the ESP32 d1 minis. Some could be flashed, but blueretro wouldn't start. The problem was the 3.3V LDO on the d1 mini board. To solve the problem you can swap the 5V LDO on the shield for a 3.3V LDO. Then you have to cut the conductor to the 5V pin and make a new connection to the 3.3V pin.
The disadvantage of converting directly from 8V to 3.3V is that the LDO gets a bit warmer. But it shouldn't be a problem.
![20210601_190531](https://user-images.githubusercontent.com/46410328/124676217-61bf1f80-debe-11eb-9770-a3355d0bb071.jpg)


The v1.2 of the shield has solder jumper for the LDO, so that you can choose whether you want to use a 5V or 3.3V LDO.
https://oshpark.com/shared_projects/aharS64j

But if your problem is really flashing:
The shield has an EN and a BOOT button. You can use this to flash as darthcloud has described it. The d1 mini board has only one EN (reset) button.

**Reply by jim150** [reply]
Thank you for your responses, I was finally able to flash both adapters and they are working perfectly on PS1 and PS2.
I love these adapters, thank you for your good work!

**Comment by Micha213** [comment]
To use the function of the IO17 LED: Simply connect IO17 to the anode of the red (power LED) with a short wire. No other changes needed. The Power LED is converted to IO17 LED.
![PXL_20210811_123951143](https://user-images.githubusercontent.com/46410328/129036948-4a0bb4de-3da5-4457-9d38-dfce82b9248e.jpg)
